### DISCUSSION TITLE
How to use `getToken` correctly after Next.js 12.2.0

### DISCUSSION THREAD

**Discussion by p-null** [nextauthjs/next-auth#5065] [Help]
Hi,

I used to rely on `getToken` to check if a user is logged in: If they are not, then redirect to login page `welcome/auth`:
```typescript
import { getToken } from "next-auth/jwt";
import { NextResponse } from "next/server";
const PUBLIC_FILE = /\.(.*)$/;

/** @param {import("next/server").NextRequest} req */
export async function middleware(req) {
    const { pathname } = req.nextUrl;

    if (
        !(
            (
                PUBLIC_FILE.test(pathname) || // exclude all files in the public folder
                pathname.startsWith("/api") || //  exclude all API routes
                pathname.includes("auth") ||
                pathname.includes("_next") ||
                pathname.startsWith("/static") // exclude static files
            ) 
        )
    ) {
        const session = await getToken({
            req,
            secret: process.env.NEXTAUTH_SECRET,
            secureCookie:
                process.env.NODE_ENV === "production" ||
                process.env.NODE_ENV === "preview",
        });

        const url = req.nextUrl.clone();
        url.pathname = "/welcome/auth";

        if (!session) return NextResponse.redirect(url);

    }
}
```
I would like to protect all pages except the login page `welcome/auth`. 

But now I found, even after a user is logged in, the `session` here is still `null`. I have read `next.js` and `next-auth` release notes, I might be missing something. Could someone remind me what is wrong?

Thanks in advance!

**Comment by mwerder** [comment]
Hey there, did you find a solution?
I am running into the exact same problem. Since I upgraded next and had to move the middleware file from _/pages/_middleware.ts_
to _/middleware.ts_ getToken() always returns _null_

**Comment by mwerder** [comment]
`next@12.2.5` seems to have fixed it! as mentioned [here](https://github.com/nextauthjs/next-auth/discussions/4969#discussioncomment-3381550)
worked for me!
